### ISSUE TITLE
Errors while building v0.8.0

### ISSUE THREAD

**Issue by KineticTactic** [DanielChappuis/reactphysics3d#153]
Errors while building the library. Here is the console output:

```
C:\Users\USER\Desktop\reactphysics3d>mingw32-make
[  1%] Building CXX object CMakeFiles/reactphysics3d.dir/src/body/CollisionBody.
cpp.obj
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/memory/MemoryManager.h:30,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:30,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/DefaultAlloca
tor.h: In member function 'virtual void reactphysics3d::DefaultAllocator::releas
e(void*, size_t)':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/DefaultAlloca
tor.h:59:52: warning: unused parameter 'size' [-Wunused-parameter]
         virtual void release(void* pointer, size_t size) override {
                                             ~~~~~~~^~~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/memory/MemoryManager.h:31,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:30,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/PoolAllocator
.h: At global scope:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/PoolAllocator
.h:106:14: error: 'mutex' in namespace 'std' does not name a type
         std::mutex mMutex;
              ^~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/PoolAllocator
.h:106:9: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '
#include <mutex>'?
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/PoolAllocator
.h:33:1:
+#include <mutex>

C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/PoolAllocator
.h:106:9:
         std::mutex mMutex;
         ^~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/memory/MemoryManager.h:32,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:30,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/HeapAllocator
.h:93:14: error: 'mutex' in namespace 'std' does not name a type
         std::mutex mMutex;
              ^~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/HeapAllocator
.h:93:9: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '#
include <mutex>'?
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/HeapAllocator
.h:35:1:
+#include <mutex>

C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/HeapAllocator
.h:93:9:
         std::mutex mMutex;
         ^~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/memory/MemoryManager.h:33,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:30,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/SingleFrameAl
locator.h:58:14: error: 'mutex' in namespace 'std' does not name a type
         std::mutex mMutex;
              ^~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/SingleFrameAl
locator.h:58:9: note: 'std::mutex' is defined in header '<mutex>'; did you forge
t to '#include <mutex>'?
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/SingleFrameAl
locator.h:1:1:
+#include <mutex>
 /******************************************************************************
**
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/memory/SingleFrameAl
locator.h:58:9:
         std::mutex mMutex;
         ^~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/utils/DebugRenderer.h:33,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsWorld.h:53,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:31,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/engine/EventListener
.h: In member function 'virtual void reactphysics3d::EventListener::onContact(co
nst reactphysics3d::CollisionCallback::CallbackData&)':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/engine/EventListener
.h:58:71: warning: unused parameter 'callbackData' [-Wunused-parameter]
         virtual void onContact(const CollisionCallback::CallbackData& callbackD
ata) override {}
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/engine/EventListener
.h: In member function 'virtual void reactphysics3d::EventListener::onTrigger(co
nst reactphysics3d::OverlapCallback::CallbackData&)':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/engine/EventListener
.h:64:69: warning: unused parameter 'callbackData' [-Wunused-parameter]
         virtual void onTrigger(const OverlapCallback::CallbackData& callbackDat
a) {}
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:32,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Sph
ereShape.h: In member function 'virtual reactphysics3d::Vector3 reactphysics3d::
SphereShape::getLocalSupportPointWithoutMargin(const reactphysics3d::Vector3&) c
onst':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Sph
ereShape.h:148:78: warning: unused parameter 'direction' [-Wunused-parameter]
 inline Vector3 SphereShape::getLocalSupportPointWithoutMargin(const Vector3& di
rection) const {
                                                               ~~~~~~~~~~~~~~~^~
~~~~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Sph
ereShape.h: In member function 'virtual bool reactphysics3d::SphereShape::testPo
intInside(const reactphysics3d::Vector3&, reactphysics3d::Collider*) const':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Sph
ereShape.h:188:79: warning: unused parameter 'collider' [-Wunused-parameter]
 inline bool SphereShape::testPointInside(const Vector3& localPoint, Collider* c
ollider) const {
                                                                     ~~~~~~~~~~^
~~~~~~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:33,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Box
Shape.h: In member function 'virtual bool reactphysics3d::BoxShape::testPointIns
ide(const reactphysics3d::Vector3&, reactphysics3d::Collider*) const':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Box
Shape.h:188:76: warning: unused parameter 'collider' [-Wunused-parameter]
 inline bool BoxShape::testPointInside(const Vector3& localPoint, Collider* coll
ider) const {
                                                                  ~~~~~~~~~~^~~~
~~~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/collision/shapes/ConcaveShape.h:31,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/collision/shapes/HeightFieldShape.h:30,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:35,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Tri
angleShape.h: In member function 'virtual reactphysics3d::Vector3 reactphysics3d
::TriangleShape::getLocalInertiaTensor(reactphysics3d::decimal) const':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Tri
angleShape.h:225:61: warning: unused parameter 'mass' [-Wunused-parameter]
 inline Vector3 TriangleShape::getLocalInertiaTensor(decimal mass) const {
                                                     ~~~~~~~~^~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Tri
angleShape.h: In member function 'virtual bool reactphysics3d::TriangleShape::te
stPointInside(const reactphysics3d::Vector3&, reactphysics3d::Collider*) const':

C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Tri
angleShape.h:230:59: warning: unused parameter 'localPoint' [-Wunused-parameter]

 inline bool TriangleShape::testPointInside(const Vector3& localPoint, Collider*
 collider) const {
                                            ~~~~~~~~~~~~~~~^~~~~~~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Tri
angleShape.h:230:81: warning: unused parameter 'collider' [-Wunused-parameter]
 inline bool TriangleShape::testPointInside(const Vector3& localPoint, Collider*
 collider) const {
                                                                       ~~~~~~~~~
~^~~~~~~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/collision/shapes/HeightFieldShape.h:30,
                 from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:35,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Con
caveShape.h: In member function 'virtual bool reactphysics3d::ConcaveShape::test
PointInside(const reactphysics3d::Vector3&, reactphysics3d::Collider*) const':
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Con
caveShape.h:133:58: warning: unused parameter 'localPoint' [-Wunused-parameter]
 inline bool ConcaveShape::testPointInside(const Vector3& localPoint, Collider*
collider) const {
                                           ~~~~~~~~~~~~~~~^~~~~~~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/collision/shapes/Con
caveShape.h:133:80: warning: unused parameter 'collider' [-Wunused-parameter]
 inline bool ConcaveShape::testPointInside(const Vector3& localPoint, Collider*
collider) const {
                                                                      ~~~~~~~~~~
^~~~~~~~
In file included from C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3
d/engine/PhysicsCommon.h:39,
                 from C:\Users\USER\Desktop\reactphysics3d\src\body\CollisionBod
y.cpp:28:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/utils/DefaultLogger.
h: At global scope:
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/utils/DefaultLogger.
h:454:14: error: 'mutex' in namespace 'std' does not name a type
         std::mutex mMutex;
              ^~~~~
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/utils/DefaultLogger.
h:454:9: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '#
include <mutex>'?
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/utils/DefaultLogger.
h:38:1:
+#include <mutex>
 #include <mutex>
C:/Users/USER/Desktop/reactphysics3d/include/reactphysics3d/utils/DefaultLogger.
h:454:9:
         std::mutex mMutex;
         ^~~
mingw32-make[2]: *** [CMakeFiles\reactphysics3d.dir\build.make:83: CMakeFiles/re
actphysics3d.dir/src/body/CollisionBody.cpp.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:95: CMakeFiles/reactphysics3d.dir/all
] Error 2
mingw32-make: *** [Makefile:160: all] Error 2
```

I generated the makefiles with CMake  using the default settings.
g++ version is 8.1.0

Note - Version 0.7.1 builds fine for me (even though there are some warnings).

**Comment by DanielChappuis** [comment]
That's strange because in all those files there is an 

`#include <mutex>`

directive at the top of the file. You said your are using the default CMake of ReactPhysics3D with the project. Therefore, you probably already compile with C++ 11 and therefore there should be no error.

As you can see [here](https://travis-ci.org/github/DanielChappuis/reactphysics3d/jobs/693561027): the library correctly builds with g++ version 8.4.0 in the Travis CI (Continuous Integration platform for ReactPhysics3D).

I am not sure what could be wrong.

**Comment by KineticTactic** [comment]
>That's strange because in all those files there is an
>`#include \<mutex\>`
>directive at the top of the file.

Yea, I see them, and I am puzzled as to why it happens. 
I also came across [this](https://stackoverflow.com/questions/14191566/c-mutex-in-namespace-std-does-not-name-a-type) StackOverflow question, which seems to suggest something about this.
It is most likely me doing something wrong, but I just don't get it.

**Comment by KineticTactic** [comment]
I am sorry for the late update.....
Ok, so I think there is some problem with my version of mingw-w64 or the installation or something. I don't think there is any problem with the library.
Even a simple piece code like the following outputs an error.
```
#include <mutex>

int main(int argc, char *argv[]) { 
    std::mutex m; 
}
```
Outputs: 
```
$ mingw32-make
g++    -c -o src/main.o src/main.cpp
src/main.cpp: In function 'int main(int, char**)':
src/main.cpp:3:41: error: 'mutex' is not a member of 'std'
 int main(int argc, char *argv[]) { std::mutex m; }
                                         ^~~~~
src/main.cpp:3:41: note: 'std::mutex' is defined in header '<mutex>'; did you forget to '#include <mutex>'?
+#include <mutex>

 int main(int argc, char *argv[]) { std::mutex m; }
                                         ^~~~~
mingw32-make: *** [<builtin>: src/main.o] Error 1
```
You may want to close this issue if you think this is not relevant.
I will update you on my progress if I get it to work.
Thanks.

**Comment by KineticTactic** [comment]
Hey, I am sorry for the late update.
But I finally got it working... finaaaally.
I came across this [StackOverflow thread](https://stackoverflow.com/questions/17242516/mingw-w64-threads-posix-vs-win32) which said that multithreading is only available on posix threads.
So I uninstalled my current installation and installed it again with posix threads option selected.
And so it finally worked! (i am so happy now :D)
Hopefully this helps if somebody is facing the same problem.
Also thanks to @DanielChappuis for creating this awesome library.

**Comment by KineticTactic** [comment]
Hey, I am sorry for the late update.
But I finally got it working... finaaaally.
I came across [this StackOverflow thread](https://stackoverflow.com/questions/17242516/mingw-w64-threads-posix-vs-win32) which said that multithreading is only available on posix threads.
So I uninstalled my current installation and installed it again with posix threads option selected.
And so it finally worked! (i am so happy now :D)
Hopefully this helps if somebody is facing the same problem.
Also thanks to @DanielChappuis for creating this awesome library.

I am closing this issue now.
